infra/notifications/src/dupeTools.js

```diff
@@ -0,0 +1,33 @@
+const web3Utils = require('web3-utils')
+const NotificationLog = require('./models').NotificationLog
+const Sequelize = require('sequelize')
+const Op = Sequelize.Op
+
+function getMessageFingerprint(messageObject) {
+  return web3Utils.keccak256(JSON.stringify(messageObject))
+}
+
+async function isNotificationDupe(messageFingerprint, config) {
+  return NotificationLog.count({
+    where: {
+      messageFingerprint: messageFingerprint,
+      sendDate: {
+        [Op.gte]: new Date(Date.now() - config.dupeLookbackMs)
+      }
+    }
+  })
+}
+
+async function logNotificationSent(messageFingerprint, ethAddress, channel) {
+  return NotificationLog.create({
+    messageFingerprint: messageFingerprint,
+    ethAddress: ethAddress,
```

Comment:
Are all those eth addresses lower case when writing/reading from DB ?
Same comment for all DB reads/writes in this PR

Reply:
Ah shoot, I of all people should have been checking that. Will do. 
Also realize I should convert them to the checksum'ed format when displaying, e.g. in an email.